ast/ddl.go

```diff
@@ -363,7 +363,30 @@ type IndexOption struct {
 
 // Restore implements Node interface.
 func (n *IndexOption) Restore(ctx *RestoreCtx) error {
-	return errors.New("Not implemented")
+	if n.KeyBlockSize > 0 {
+		ctx.WriteKeyWord("KEY_BLOCK_SIZE")
+		ctx.WritePlain("=")
+		ctx.WritePlainf("%d", n.KeyBlockSize)
+	}
+
+	if n.KeyBlockSize > 0 && n.Tp != model.IndexTypeInvalid {
```

Comment:
🤔 This will repeat the condition 3 times. How about encode this in a variable?

```go
hasPrevOption := false 
if n.KeyBlockSize > 0 {
    write(...)
    hasPrevOption = true
}
if n.Tp != model.IndexTypeInvalid {
    if hasPrevOption {
        ctx.WritePlain(" ")
    }
    write(...)
    hasPrevOption = true
}
if n.Comment != "" {
    if hasPrevOption {
        ctx.WritePlain(" ")
    }
    write(...)
}
```

Reply:
I see what you mean. Let me see